### PR TITLE
Acr fix

### PR DESCRIPTION
**Pull request by VeryEarly**
## Description

Connect-AzContainerRegistry login to ACR using access token in version 2.2.0 and refresh token before.

Bug was found images cannot be pushed after login.

Switch back to refresh token for login.

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/ContainerRegistry/ContainerRegistry/Models/ContainerRegistryDataPlaneClient.cs [hunk 1]
```diff
@@ -123,7 +123,7 @@ private string GetRefreshToken()
                     .RefreshTokenProperty;
         }
 
-        private string GetAccessToken(string scope)
+        public string GetAccessToken(string scope)
```
Comment (dingmeng-xue) [inline src/ContainerRegistry/ContainerRegistry/Models/ContainerRegistryDataPlaneClient.cs:126]: Do you need to change the scope of this method?

File: tools/Az/Az.psd1 [hunk 2]
```diff
@@ -66,7 +66,7 @@ RequiredModules = @(@{ModuleName = 'Az.Accounts'; ModuleVersion = '2.2.5'; },
                @{ModuleName = 'Az.CognitiveServices'; RequiredVersion = '1.8.0'; }, 
                @{ModuleName = 'Az.Compute'; RequiredVersion = '4.9.0'; }, 
                @{ModuleName = 'Az.ContainerInstance'; RequiredVersion = '1.0.3'; }, 
-               @{ModuleName = 'Az.ContainerRegistry'; RequiredVersion = '2.2.0'; }, 
+               @{ModuleName = 'Az.ContainerRegistry'; RequiredVersion = '2.2.1'; },
```
Comment (dingmeng-xue) [inline tools/Az/Az.psd1:69]: Please check if you need to change the version in Az. 

Try below steps
1. Keep all changes except reverting Az.psd1.
2. `dotnet msbuild build.proj /t:"Build" /p:"Configuration=Release"`
3. `.\RunVersionController.ps1 -Release "February 2021"`

I'd like to know the result of Az.psd1.  Can it pick up the latest version of ContainerRegistry? Can it add its release note to release note of Az?

File: src/ContainerRegistry/ContainerRegistry/Models/ContainerRegistryDataPlaneClient.cs (hunk 1 above)
Comment (VeryEarly) [reply src/ContainerRegistry/ContainerRegistry/Models/ContainerRegistryDataPlaneClient.cs:126]: > Do you need to change the scope of this method?

Cli provided a way to expose this access token in cmdlet, though we do not have it right now. It's possible we need it in the future.

File: tools/Az/Az.psd1 (hunk 2 above)
Comment (VeryEarly) [reply tools/Az/Az.psd1:69]: > Please check if you need to change the version in Az.
> 
> Try below steps
> 
> 1. Keep all changes except reverting Az.psd1.
> 2. `dotnet msbuild build.proj /t:"Build" /p:"Configuration=Release"`
> 3. `.\RunVersionController.ps1 -Release "February 2021"`
> 
> I'd like to know the result of Az.psd1. Can it pick up the latest version of ContainerRegistry? Can it add its release note to release note of Az?

After tried these steps, version of Az.ContainerRegistry in Az.psd1 is still 2.2.0

File: src/ContainerRegistry/ContainerRegistry/Models/ContainerRegistryDataPlaneClient.cs (hunk 1 above)
Comment (dingmeng-xue) [reply src/ContainerRegistry/ContainerRegistry/Models/ContainerRegistryDataPlaneClient.cs:126]: Please public it when you use it. Don't add too much in one PR.

**Comment by dingmeng-xue** [comment]
Please update subject of this pull request.

**Comment by dingmeng-xue** [comment]
> ## Description
> Connect-AzContainerRegistry login to ACR using access token in version 2.2.0 and refresh token before.
> 
> Bug was found images cannot be pushed after login.
> 
> Switch back to refresh token for login.

Normally, refresh token is used to redeem the access token only. Access token is used for accessing resource. Does ACR use different term? Please confirm. Is there doc to explain it?

**Comment by VeryEarly** [comment]
> > ## Description
> > Connect-AzContainerRegistry login to ACR using access token in version 2.2.0 and refresh token before.
> > Bug was found images cannot be pushed after login.
> > Switch back to refresh token for login.
> 
> Normally, refresh token is used to redeem the access token only. Access token is used for accessing resource. Does ACR use different term? Please confirm. Is there doc to explain it?

I do not find any doc about it. The Cli implementation is using refresh token to do login (what we also used before this release)

**Pull request merged**